lib/https.js

```diff
@@ -58,7 +58,22 @@ function createConnection(port, host, options) {
   }
 
   debug('createConnection', options);
-  return tls.connect(options);
+
+  var self = this;
+  if (options._agentKey && this._tlsSessions[options._agentKey]) {
+    debug('reuse session for %j', options._agentKey);
+    options = util._extend({
+      session: this._tlsSessions[options._agentKey]
```

Comment:
I think we [already make a copy](https://github.com/nodejs/io.js/blob/eb35968de7a040d737f3280e9fed0b6edf566b2b/lib/_http_client.js#L26), that's why I'm asking. :) Not trying to be a stickler - you could say that call and I [go back a little bit](https://github.com/nodejs/io.js/pull/1467).